### PR TITLE
[TEP-0079] Catalog Automated Testing and Tekton Dogfooding

### PR DESCRIPTION
**Pull request by QuanZhang-William**
This commit updates the "Automated Testing and Dogfooding" section of TEP-0079 based on the [previous work](https://hackmd.io/80AHmjpWQMixaP3-qkXyww?edit), which adds the testing strategy proposal for the Verified Catalogs and a testing infra template for Community Catalogs.

This commit also update the tep from `proposed` to `implementable` 🥂

### CODE REVIEW

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Comment by pritidesai** [comment]
@vinamra28 please approve the proposal instead of `lgtm` based on our guidelines.

File: teps/0079-tekton-catalog-support-tiers.md
Comment (dibyom) [inline teps/0079-tekton-catalog-support-tiers.md:541]: nit: the user is providing a pipeline but the file is called run.yaml?

File: teps/0079-tekton-catalog-support-tiers.md
Comment (dibyom) [inline teps/0079-tekton-catalog-support-tiers.md:715]: Are we suggesting each project have their own eventlistener?

File: teps/0079-tekton-catalog-support-tiers.md
Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md:541]: Good point, we use `run.yaml` today but the naming can be improved for sure. Let's call it `pipelineRun.yaml`？

Comment (jerop) [inline teps/0079-tekton-catalog-support-tiers.md:541]:
```suggestion
Following the Catalog Contract defined in [TEP-0115][tep-0115], the owners need to provide one or more test `pipelines` in the `./{resource-type}/{resource-name}/tests/run.yaml` file of the catalog resource directories to run the integration test for the resource.
```

Comment (jerop) [reply teps/0079-tekton-catalog-support-tiers.md:541]: nit: please format the other links as shown above -- to be consistent with the rest of the TEP and make it easier to read

File: teps/0079-tekton-catalog-support-tiers.md
Comment (jerop) [inline teps/0079-tekton-catalog-support-tiers.md:658]:
```suggestion
To dogfood Tekton, we propose using Tekton as the test automation infrastructure for the Verified Catalogs, leveraging the [Dogfooding Cluster](https://github.com/tektoncd/plumbing/blob/main/docs/dogfooding.md) and the existing [Tekton setup](https://github.com/tektoncd/plumbing/blob/main/tekton/README.md) in the plumbing repo.
```

**Review by jerop — COMMENTED** [review]
Thank you @QuanZhang-William!

There are some open questions around the testing strategy:
- how often would we run the integration tests? nightly? on PRs only?
- which Tekton versions would we test these resources against? nightly release? latest release? minimum compatible version?

@bobcatfish discussed some ideas here: https://github.com/tektoncd/community/pull/170

/assign

**Comment by jerop** [comment]
putting a hold so that it's not accidentally merged on the first approval because it already has an lgtm

/hold

File: teps/0079-tekton-catalog-support-tiers.md
Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md]: Hi @dibyom, I have changed it to `pipelineRun.yaml`. PTAL~

Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md]: Thanks for the feedback! I have update the format across the file. PTAL.

File: teps/0079-tekton-catalog-support-tiers.md
Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md]: Thanks, updated!

**Comment by QuanZhang-William** [comment]
> Thank you @QuanZhang-William!
> 
> There are some open questions around the testing strategy:
> 
> * how often would we run the integration tests? nightly? on PRs only?
> * which Tekton versions would we test these resources against? nightly release? latest release? minimum compatible version?
> 
> @bobcatfish discussed some ideas here: #170
> 
> /assign

Thanks @jerop for the link! I have added 2 sections under the [Integration Test](https://github.com/tektoncd/community/blob/60d762120fb2a33a9aa3e3b129e353b5ec034e51/teps/0079-tekton-catalog-support-tiers.md#integration-test) section, talking about the nightly integration test and PR merge integration test respectively, including the `Tekton Pipelines` release that will be used. PTAL 😁

File: teps/0079-tekton-catalog-support-tiers.md
Comment (jerop) [inline teps/0079-tekton-catalog-support-tiers.md:705]: can we send slack notifications to maintainers? wondering how the tekton pipelines maintainers can be aware of the failures in nightly releases that are caught in the nightly catalog tests

Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md]: SGTM! I have updated this section. PTAL!

**Review by jerop — APPROVED** [review]
Thank you @QuanZhang-William!

**Comment by jerop** [comment]
/hold cancel

@dibyom @vinamra28 please take a look

File: teps/0079-tekton-catalog-support-tiers.md
Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md:898]: SG! Updated.

File: teps/0079-tekton-catalog-support-tiers.md
Comment (QuanZhang-William) [reply teps/0079-tekton-catalog-support-tiers.md]: SGTM, updated!

**Review by dibyom — APPROVED** [review]

**Comment by afrittoli** [comment]
Approved during API WG
/lgtm

**Pull request merged**
